packages/@lwc/engine-dom/src/styles.ts

```diff
@@ -145,13 +140,14 @@ function doInsertStylesheet(content: string, target: ShadowRoot | Document, cach
     }
 }
 
-function getCacheData(content: string) {
+function getCacheData(content: string, forceStyleElement: boolean) {
     let cacheData = stylesheetCache.get(content);
```

Comment:
Something looks fishy here. Shouldn't the cache key also include the `forStyleElement`? 

Currently, you can find yourself in a situation where `insertStyleElement` at line 170 where the `cacheData` is a `ConstructableStylesheetCacheData` and not a `StyleElementCacheData` (eg. `getCacheData` is invoked twice with same content but with `forceStyleElement` set to `false` first and then `true`)